status-im/status-desktop
# refactor(Wallet/SendModal): Selected Recipients component review, backend dependencies cleanup and basic test support added

### What does the PR do

Closes #15208

- Component name changed to be more precise with what it represents.
- Removed `RootStore` dependency from `Helpers.qml`.
- Removed unnecessary method `lookupAddressObject` from `RootStore`.
- Updated `SendModal` property bindings related to recipient.
- Some activity controller interaction details moved to transactions store instead of being directly in `SendModal`.
- Type enum living in `RecipientSelectorPanel` is moved to `Helpers`. Since it doesn't depends on the specific component and it's used in other global places.
- Created data transformation file called `WalletAccountsAdaptor` for `SendModal`. Now it contains the owned wallet accounts model data transformations.
- Added common formatting method for short chain ids to some utils and partially removed backend dependency.
- Added new page for `RecipientSelectorPanelPage` in storybook
- Just some basic test cases in a new created `tst_RecipientsSelectorPanel` file. It needs a second step to add more interesting coverage.

### Affected areas

**_SendModal_**

### Impact on end user

No impact. No changes to the end user, just code improvements and initial component test support.

### How to test

Flows related to the interaction with the `Recipients Selector` component:
- Select **SharedAddress tab** / item:
  - See the selected item is correctly selected
  - See the expected data is visualized in the list

- Same checks for **my accounts tab** and **recent tabs**.
- TODO: In case of `recent` tab selected, ensure that when account is changed, it triggers a reload (now broken bc of rebase)
- Ensure the recipients selector panel is visualized and correcly behaving from different entry points (when triggering the send modal from right click on selected asset, from a recent transaction, etc)

### Risk
